README.md

```diff
@@ -1,88 +1,86 @@
+   (currently only the CPS data set includes a benefits file).
 
 Note that the factors file is the same in both sets of data files
 because the variable extrapolation factors are independent of the
-sample data being used.  But the weights and ratios files do depend on
-the data file, so they are different in the two sets of data files.
-
-
-Data-Preparation Documentation
-------------------------------
-
-**IRS-SOI Public Use File (PUF)** documentation:
-
-1. [PUF-based sample data](puf_data/README.md);
-
-2. [grow factors](stage1/README.md)
-
-3. [PUF-based sample weights](puf_stage2/README.md);
-
-4. [PUF-based adjustment ratios](puf_stage3/README.md).
-
-**Census Current Population Survey (CPS)** documentation is available here:
-
-1. [CPS-based sample data](cps_data/README.md);
-
-2. [grow factors](stage1/README.md)
-
-3. [CPS-based sample weights](cps_stage2/README.md);
-
-4. [CPS-based adjustment ratios](cps_stage3/README.md).
-
-
-Contributing
-------------
-Before opening up a pull request, run `pytest` to ensure all tests pass.
-
-
-Work-Flow Documentation
------------------------
-
-The sequence of operations required to make the two sets of data files
-is contained in the [`csvmake` bash script](csvmake), which also
-automates the preparation work-flow (except on Windows).
-
-The sequence of operations required to install the two sets of data
-files in the Tax-Calculator repository is contained in the [`csvcopy`
-bash script](csvcopy), which also automates the installation work-flow
-(except on Windows).
+sample data being used.  But the weights, ratios, and benefits files
+do depend on the data file, so they are different in the two sets of
+data files.
+
+
+Data-Preparation Documentation and Workflow
+-------------------------------------------
+
+The best documentation of the data-preparation workflow is the
+[taxdata Makefile](Makefile).  The Makefile shows the input files and
+the Python script that generates each made file.  The files made in
+early stages of the workflow serve as input files in later stages,
+which means there is a cascading effect of changes in the scripts
+and/or input files.  The Makefile handles all this complexity in an
+economical way because it executes scripts to make new versions of
+made files only when necessary.  Start exploring the Makefile by
+running the `make help` command.
+
+Note that the stage2 linear program that generates the weights file is
+very long-running, taking five or more hours depending on your
+computer's CPU speed.  We are considering options for speeding up this
+stage2 work, but for the time being you can execute `make puf-files`
+and `make cps-files` in separate terminal windows to have the two
+stage2 linear programs run in parallel.  If you try this, be sure to
+wait a bit until the `make puf-files` starts on stage2 before running
+the `make cps-files` command in the other terminal window.  If you are
```

Comment:
Out of curiosity, is there a reason they should wait before running `make cps-files`? Does it have something to do with how makefiles work or did you run into a problem running `make puf-files` and `make cps-files` at the same time?

Reply:
It's because `make cps-files` and `make puf-files` both generate the intermediate factors files.
It seems as if running them simultaneously might cause confusion with the date-stamp on the made files.  Doing one first (which takes just a moment because stage1 is fast) means that when you start the second, the Makefile will skip stage1.  Does this make sense?

Reply:
Yes, that makes sense. Thanks for the clarification.